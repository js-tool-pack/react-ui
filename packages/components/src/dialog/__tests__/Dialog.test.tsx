import { fireEvent, render } from '@testing-library/react';
import Demo from '../demo/basic';
import { Dialog } from '..';

describe('Dialog', () => {
  const getRoot = () => document.querySelector('.t-dialog__root');

  test('basic', () => {
    const { container } = render(<Demo />);
    expect(document.body).toMatchSnapshot();
    fireEvent.click(container.querySelector('.t-button')!);
    expect(getRoot()).toMatchSnapshot();
  });

  test('attrs', () => {
    render(
      <Dialog
        bodyAttrs={{ className: 'bar' }}
        attrs={{ className: 'foo' }}
        visible
        center
      />,
    );
    expect(getRoot()).toMatchSnapshot();
  });

  test('center', () => {
    render(<Dialog visible center />);
    expect(getRoot()!.querySelector('.t-dialog')).toHaveClass(
      't-dialog__center',
    );
  });

  test('centered', () => {
    render(<Dialog centered visible />);
    expect(getRoot()!.querySelector('.t-dialog__wrapper')).toHaveClass(
      't-dialog__centered',
    );
  });

  describe('closeOnClickMask', () => {
    const getMask = () => document.querySelector('.t-dialog__mask');

    test('off', () => {
      render(<Dialog closeOnClickMask={false} visible />);
      fireEvent.click(getMask()!);
      expect(getRoot()).not.toHaveClass('t-dialog-leave-active');
    });

    test('on onClose', () => {
      const onClose = jest.fn();
      expect(onClose).not.toBeCalled();
      render(<Dialog onClose={onClose} closeOnClickMask visible />);
      fireEvent.click(getMask()!);
      expect(getRoot()).toHaveClass('t-dialog-leave-active');
      expect(onClose).toBeCalled();
    });
  });

  describe('esc', () => {
    const escKeydown = () => fireEvent.keyDown(window, { key: 'Escape' });

    test('off', () => {
      render(<Dialog esc={false} visible />);
      escKeydown();
      expect(getRoot()).not.toHaveClass('t-dialog-leave-active');
    });

    test('on onClose', () => {
      const onClose = jest.fn();
      expect(onClose).not.toBeCalled();
      render(<Dialog onClose={onClose} visible esc />);
      escKeydown();
      expect(getRoot()).toHaveClass('t-dialog-leave-active');
      expect(onClose).toBeCalled();
    });
  });

  test('zIndex', () => {
    render(<Dialog zIndex={12345} visible />);
    expect(getRoot()).toHaveStyle({ zIndex: 12345 });
  });

  describe('header', () => {
    test('content', () => {
      render(<Dialog header="123" visible />);
      const root = getRoot();
      expect(root).toMatchSnapshot();
      expect(root!.querySelector('.t-dialog__header')).not.toBeNull();
    });
    test('null', () => {
      render(<Dialog header={null} visible />);
      const root = getRoot();
      expect(root).toMatchSnapshot();
      expect(root!.querySelector('.t-dialog__header')).toBeNull();
    });
  });

  describe('footer', () => {
    test('content', () => {
      render(<Dialog footer="123" visible />);
      const root = getRoot();
      expect(root).toMatchSnapshot();
      expect(root!.querySelector('.t-dialog__footer')).not.toBeNull();
    });
    test('null', () => {
      render(<Dialog footer={null} visible />);
      const root = getRoot();
      expect(root).toMatchSnapshot();
      expect(root!.querySelector('.t-dialog__footer')).toBeNull();
    });
  });
});
