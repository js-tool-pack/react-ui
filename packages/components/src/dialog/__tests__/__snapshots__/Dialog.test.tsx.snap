// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog attrs 1`] = `
<div
  class="t-dialog__root foo"
  style="z-index: 1000;"
>
  <div
    class="t-dialog__mask"
  />
  <div
    class="t-dialog__wrapper"
  >
    <section
      class="t-layout t-dialog bar t-dialog__center t-layout--v"
      style="transform-origin: calc(50% + 0px) calc(0px - 50%);"
    >
      <header
        class="t-header t-dialog__header"
      >
        <span
          class="t-dialog__title"
        />
        <button
          class="t-button t-dialog__btn-close t-button--type-default t--size-sm t-button--plain-text t-button--icon-l"
        >
          <span>
            <i
              class="t-icon"
              role="img"
            >
              <svg
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M574.56865938 513.72328935l366.30945353-364.40941657c17.3212673-17.23289347 17.4096411-45.15901828 0.17674762-62.48028556-17.27708037-17.36545419-45.15901828-17.4096411-62.48028556-0.17674764l-366.39782734 364.45360348-363.21637009-364.32104275C131.77167097 69.55650684 103.80135925 69.46813303 86.48009197 86.70102651 69.20301159 103.93391997 69.15882467 131.90423169 86.39171815 149.22549897l363.12799629 364.23266894L83.38700853 877.64664993c-17.3212673 17.23289347-17.4096411 45.15901828-0.17674763 62.48028558 8.61644674 8.66063363 19.97248167 13.03513737 31.32851659 13.03513736 11.26766112 0 22.53532223-4.28612991 31.15176897-12.85838974l366.22107973-364.27685584 366.53038806 367.63506071c8.61644674 8.66063363 19.92829476 12.99095045 31.2843297 12.99095047 11.31184801 0 22.57950913-4.33031682 31.19595588-12.90257666 17.27708037-17.23289347 17.3212673-45.15901828 0.0883738-62.48028554L574.56865938 513.72328935z"
                  fill="currentColor"
                />
              </svg>
            </i>
          </span>
        </button>
      </header>
      <main
        class="t-main t-dialog__main"
      />
      <footer
        class="t-footer t-dialog__footer"
      />
    </section>
  </div>
</div>
`;

exports[`Dialog basic 1`] = `
<body>
  <div>
    <div
      style="text-align: center;"
    >
      <button
        class="t-button t-button--type-primary t--size-m t-button--icon-l"
      >
        <span>
          打开
          dialog
        </span>
      </button>
    </div>
  </div>
</body>
`;

exports[`Dialog basic 2`] = `
<div
  class="t-dialog__root t-dialog-enter-active t-dialog-enter-to"
  style="z-index: 1000;"
>
  <div
    class="t-dialog__mask"
  />
  <div
    class="t-dialog__wrapper"
  >
    <section
      class="t-layout t-dialog t-layout--v"
      style="top: 100px; transform-origin: calc(50% + 0px) calc(0px - 100px);"
    >
      <header
        class="t-header t-dialog__header"
      >
        <span
          class="t-dialog__title"
        >
          title
        </span>
        <button
          class="t-button t-dialog__btn-close t-button--type-default t--size-sm t-button--plain-text t-button--icon-l"
        >
          <span>
            <i
              class="t-icon"
              role="img"
            >
              <svg
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M574.56865938 513.72328935l366.30945353-364.40941657c17.3212673-17.23289347 17.4096411-45.15901828 0.17674762-62.48028556-17.27708037-17.36545419-45.15901828-17.4096411-62.48028556-0.17674764l-366.39782734 364.45360348-363.21637009-364.32104275C131.77167097 69.55650684 103.80135925 69.46813303 86.48009197 86.70102651 69.20301159 103.93391997 69.15882467 131.90423169 86.39171815 149.22549897l363.12799629 364.23266894L83.38700853 877.64664993c-17.3212673 17.23289347-17.4096411 45.15901828-0.17674763 62.48028558 8.61644674 8.66063363 19.97248167 13.03513737 31.32851659 13.03513736 11.26766112 0 22.53532223-4.28612991 31.15176897-12.85838974l366.22107973-364.27685584 366.53038806 367.63506071c8.61644674 8.66063363 19.92829476 12.99095045 31.2843297 12.99095047 11.31184801 0 22.57950913-4.33031682 31.19595588-12.90257666 17.27708037-17.23289347 17.3212673-45.15901828 0.0883738-62.48028554L574.56865938 513.72328935z"
                  fill="currentColor"
                />
              </svg>
            </i>
          </span>
        </button>
      </header>
      <main
        class="t-main t-dialog__main"
      >
        <div>
          随便说说
        </div>
        <textarea
          placeholder="输入..."
          style="margin-top: 10px; display: block; width: 100%; box-sizing: border-box;"
        />
      </main>
      <footer
        class="t-footer t-dialog__footer"
      >
        footer
      </footer>
    </section>
  </div>
</div>
`;

exports[`Dialog footer content 1`] = `
<div
  class="t-dialog__root"
  style="z-index: 1000;"
>
  <div
    class="t-dialog__mask"
  />
  <div
    class="t-dialog__wrapper"
  >
    <section
      class="t-layout t-dialog t-layout--v"
      style="transform-origin: calc(50% + 0px) calc(0px - 50%);"
    >
      <header
        class="t-header t-dialog__header"
      >
        <span
          class="t-dialog__title"
        />
        <button
          class="t-button t-dialog__btn-close t-button--type-default t--size-sm t-button--plain-text t-button--icon-l"
        >
          <span>
            <i
              class="t-icon"
              role="img"
            >
              <svg
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M574.56865938 513.72328935l366.30945353-364.40941657c17.3212673-17.23289347 17.4096411-45.15901828 0.17674762-62.48028556-17.27708037-17.36545419-45.15901828-17.4096411-62.48028556-0.17674764l-366.39782734 364.45360348-363.21637009-364.32104275C131.77167097 69.55650684 103.80135925 69.46813303 86.48009197 86.70102651 69.20301159 103.93391997 69.15882467 131.90423169 86.39171815 149.22549897l363.12799629 364.23266894L83.38700853 877.64664993c-17.3212673 17.23289347-17.4096411 45.15901828-0.17674763 62.48028558 8.61644674 8.66063363 19.97248167 13.03513737 31.32851659 13.03513736 11.26766112 0 22.53532223-4.28612991 31.15176897-12.85838974l366.22107973-364.27685584 366.53038806 367.63506071c8.61644674 8.66063363 19.92829476 12.99095045 31.2843297 12.99095047 11.31184801 0 22.57950913-4.33031682 31.19595588-12.90257666 17.27708037-17.23289347 17.3212673-45.15901828 0.0883738-62.48028554L574.56865938 513.72328935z"
                  fill="currentColor"
                />
              </svg>
            </i>
          </span>
        </button>
      </header>
      <main
        class="t-main t-dialog__main"
      />
      <footer
        class="t-footer t-dialog__footer"
      >
        123
      </footer>
    </section>
  </div>
</div>
`;

exports[`Dialog footer null 1`] = `
<div
  class="t-dialog__root"
  style="z-index: 1000;"
>
  <div
    class="t-dialog__mask"
  />
  <div
    class="t-dialog__wrapper"
  >
    <section
      class="t-layout t-dialog t-layout--v"
      style="transform-origin: calc(50% + 0px) calc(0px - 50%);"
    >
      <header
        class="t-header t-dialog__header"
      >
        <span
          class="t-dialog__title"
        />
        <button
          class="t-button t-dialog__btn-close t-button--type-default t--size-sm t-button--plain-text t-button--icon-l"
        >
          <span>
            <i
              class="t-icon"
              role="img"
            >
              <svg
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M574.56865938 513.72328935l366.30945353-364.40941657c17.3212673-17.23289347 17.4096411-45.15901828 0.17674762-62.48028556-17.27708037-17.36545419-45.15901828-17.4096411-62.48028556-0.17674764l-366.39782734 364.45360348-363.21637009-364.32104275C131.77167097 69.55650684 103.80135925 69.46813303 86.48009197 86.70102651 69.20301159 103.93391997 69.15882467 131.90423169 86.39171815 149.22549897l363.12799629 364.23266894L83.38700853 877.64664993c-17.3212673 17.23289347-17.4096411 45.15901828-0.17674763 62.48028558 8.61644674 8.66063363 19.97248167 13.03513737 31.32851659 13.03513736 11.26766112 0 22.53532223-4.28612991 31.15176897-12.85838974l366.22107973-364.27685584 366.53038806 367.63506071c8.61644674 8.66063363 19.92829476 12.99095045 31.2843297 12.99095047 11.31184801 0 22.57950913-4.33031682 31.19595588-12.90257666 17.27708037-17.23289347 17.3212673-45.15901828 0.0883738-62.48028554L574.56865938 513.72328935z"
                  fill="currentColor"
                />
              </svg>
            </i>
          </span>
        </button>
      </header>
      <main
        class="t-main t-dialog__main"
      />
    </section>
  </div>
</div>
`;

exports[`Dialog header content 1`] = `
<div
  class="t-dialog__root"
  style="z-index: 1000;"
>
  <div
    class="t-dialog__mask"
  />
  <div
    class="t-dialog__wrapper"
  >
    <section
      class="t-layout t-dialog t-layout--v"
      style="transform-origin: calc(50% + 0px) calc(0px - 50%);"
    >
      <header
        class="t-header t-dialog__header"
      >
        <span
          class="t-dialog__title"
        >
          123
        </span>
        <button
          class="t-button t-dialog__btn-close t-button--type-default t--size-sm t-button--plain-text t-button--icon-l"
        >
          <span>
            <i
              class="t-icon"
              role="img"
            >
              <svg
                viewBox="0 0 1024 1024"
              >
                <path
                  d="M574.56865938 513.72328935l366.30945353-364.40941657c17.3212673-17.23289347 17.4096411-45.15901828 0.17674762-62.48028556-17.27708037-17.36545419-45.15901828-17.4096411-62.48028556-0.17674764l-366.39782734 364.45360348-363.21637009-364.32104275C131.77167097 69.55650684 103.80135925 69.46813303 86.48009197 86.70102651 69.20301159 103.93391997 69.15882467 131.90423169 86.39171815 149.22549897l363.12799629 364.23266894L83.38700853 877.64664993c-17.3212673 17.23289347-17.4096411 45.15901828-0.17674763 62.48028558 8.61644674 8.66063363 19.97248167 13.03513737 31.32851659 13.03513736 11.26766112 0 22.53532223-4.28612991 31.15176897-12.85838974l366.22107973-364.27685584 366.53038806 367.63506071c8.61644674 8.66063363 19.92829476 12.99095045 31.2843297 12.99095047 11.31184801 0 22.57950913-4.33031682 31.19595588-12.90257666 17.27708037-17.23289347 17.3212673-45.15901828 0.0883738-62.48028554L574.56865938 513.72328935z"
                  fill="currentColor"
                />
              </svg>
            </i>
          </span>
        </button>
      </header>
      <main
        class="t-main t-dialog__main"
      />
      <footer
        class="t-footer t-dialog__footer"
      />
    </section>
  </div>
</div>
`;

exports[`Dialog header null 1`] = `
<div
  class="t-dialog__root"
  style="z-index: 1000;"
>
  <div
    class="t-dialog__mask"
  />
  <div
    class="t-dialog__wrapper"
  >
    <section
      class="t-layout t-dialog t-layout--v"
      style="transform-origin: calc(50% + 0px) calc(0px - 50%);"
    >
      <main
        class="t-main t-dialog__main"
      />
      <footer
        class="t-footer t-dialog__footer"
      />
    </section>
  </div>
</div>
`;
